app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -61,27 +53,15 @@ ManageIQ.angular.app.controller('cloudSubnetFormController', ['$http', '$scope',
 
   $scope.filterNetworkManagerChanged = function(id) {
     miqService.sparkleOn();
-
-    $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id)
-      .then(getCloudSubnetNetworksByEmsComplete)
-      .catch(miqService.handleFailure);
-
-    $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id)
-      .then(getCloudTenantsByEmsComplete)
-      .catch(miqService.handleFailure);
-
+    if (id) {
+      API.get("/api/cloud_networks?expand=resources&attributes=name,ems_ref&filter[]=ems_id=" + id).then(function(data) {
```

Comment:
@himdel : I wonder if we should have some wrapper around the API calls. Something that would construct the URL at least.

SUI's https://github.com/ManageIQ/manageiq-ui-service/blob/master/client/app/core/collections-api.factory.js comes to my mind.